go.mod

```diff
@@ -94,9 +95,11 @@ replace (
 	// TODO: Remove this: https://github.com/thanos-io/thanos/issues/3967.
 	github.com/minio/minio-go/v7 => github.com/bwplotka/minio-go/v7 v7.0.11-0.20210324165441-f9927e5255a6
 	// Make sure Prometheus version is pinned as Prometheus semver does not include Go APIs.
-	github.com/prometheus/prometheus => github.com/prometheus/prometheus v1.8.2-0.20211101135822-b862218389fc
+	// github.com/prometheus/prometheus => github.com/prometheus/prometheus v1.8.2-0.20211119115433-692a54649ed7
+	// Custom fork with missing packages (testing, before upstreaming).
+	github.com/prometheus/prometheus => github.com/pstibrany/prometheus v1.8.2-0.20211119134801-de51b7b3200a
 	github.com/sercand/kuberesolver => github.com/sercand/kuberesolver v2.4.0+incompatible
-	google.golang.org/grpc => google.golang.org/grpc v1.29.1
+	// google.golang.org/grpc => google.golang.org/grpc v1.29.1
```

Comment:
Is this not necessary anymore?

Reply:
With pin to v1.29.1 the code no longer compiles. I can move the pin to v1.40.0 if that's what you want.

Reply:
Yes please.

Reply:
OK, done.